pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -66,8 +71,105 @@ def setup_environment(self) -> None:
         environment before setup is complete.
         """
 
-    def setup(self) -> None:
-        """Called by the accelerator to finish setup."""
+    def setup_optimizers(self, trainer: "pl.Trainer") -> None:
+        """Creates optimizers and schedulers.
+
+        Args:
+            trainer: the Trainer, these optimizers should be connected to
+        """
+        if trainer.state.fn not in (TrainerFn.FITTING, TrainerFn.TUNING):
+            return
+        optimizers, lr_schedulers, optimizer_frequencies = self.init_optimizers(
+            trainer=trainer, model=self.lightning_module
+        )
+        self.optimizers = optimizers
+        self.lr_schedulers = lr_schedulers
+        self.optimizer_frequencies = optimizer_frequencies
+
+    def setup(self, trainer: "pl.Trainer") -> None:
+        """Setup plugins for the trainer fit and creates optimizers.
+
+        Args:
+            trainer: the trainer instance
+        """
+        # call super()
```

Comment:
not needed I believe

Reply:
Was a note to myself to remember call super if there is override in subclass, will remove